src/pages/Lending/Pool/components/Repay/Repay.tsx

```diff
@@ -12,27 +12,38 @@ const RepayEntries = [RepayRoutePaths.Input, RepayRoutePaths.Confirm]
 type RepayProps = {
   collateralAccountId: AccountId
   repaymentAccountId: AccountId
+  repaymentAsset: Asset | null
+  setRepaymentAsset: (asset: Asset | null) => void
   onCollateralAccountIdChange: (accountId: AccountId) => void
   onRepaymentAccountIdChange: (accountId: AccountId) => void
+  repaymentPercent: number
+  setRepaymentPercent: (value: number) => void
 }
 
 export const Repay = ({
   collateralAccountId,
   repaymentAccountId: borrowAccountId,
+  repaymentAsset,
+  setRepaymentAsset,
   onCollateralAccountIdChange: handleCollateralAccountIdChange,
   onRepaymentAccountIdChange: handleRepaymentAccountIdChange,
+  repaymentPercent,
+  setRepaymentPercent,
 }: RepayProps) => {
-  const [repaymentPercent, setRepaymentPercent] = useState<number>(100)
-
   const collateralAssetId = useRouteAssetId()
 
-  const handleDepositAmountChange = useCallback((value: number) => {
-    setRepaymentPercent(value)
-  }, [])
+  const handleDepositAmountChange = useCallback(
+    (value: number) => {
+      setRepaymentPercent(value)
+    },
+    [setRepaymentPercent],
+  )
 
   return (
     <MemoryRouter initialEntries={RepayEntries} initialIndex={0}>
       <RepayRoutes
+        repaymentAsset={repaymentAsset}
+        setRepaymentAsset={setRepaymentAsset}
         collateralAssetId={collateralAssetId}
         repaymentPercent={repaymentPercent}
         onRepaymentPercentChange={handleDepositAmountChange}
```

Comment:
might be able to remove the wrapped callback

```suggestion
  return (
    <MemoryRouter initialEntries={RepayEntries} initialIndex={0}>
      <RepayRoutes
        repaymentAsset={repaymentAsset}
        setRepaymentAsset={setRepaymentAsset}
        collateralAssetId={collateralAssetId}
        repaymentPercent={repaymentPercent}
        onRepaymentPercentChange={setRepaymentPercent}
```

Reply:
🧠  https://github.com/shapeshift/web/pull/5679/commits/a9e114696c9411e315d72ef1456aee41743a8569